### PR TITLE
add comparative between path directives

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #474
Add comparative between path directives and link to this comparative in all mentioned directives docu page

### CODE REVIEW

**Comment by jlprat** [comment]
@jrudolph let me know if this is what you had in mind for the issue #474

**Comment by jrudolph** [comment]
PLS BUILD

File: docs/src/main/paradox/java/http/routing-dsl/directives/path-directives/path.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ The `path` directive attempts to match the **complete** remaining path, not just
 a path prefix and then delegate further filtering to a lower level in your routing structure use the @ref[pathPrefix](pathPrefix.md#pathprefix-java)
 directive instead. As a consequence it doesn't make sense to nest a `path` or @ref[pathPrefix](pathPrefix.md#pathprefix-java) directive
 underneath another `path` directive, as there is no way that they will ever match (since the unmatched path underneath
-a `path` directive will always be empty).
+a `path` directive will always be empty). For a comparative between path directives check @ref[Overview of path directives](index.md#overview-path-java).
```
Comment (jrudolph) [inline docs/src/main/paradox/java/http/routing-dsl/directives/path-directives/path.md]: comparative -> comparison (everywhere ;) )

**Comment by jrudolph** [comment]
LGTM. Great, thanks, @jlprat. I think this is a great start. (It would be nice to have a table with examples as well but this can be added in another PR.)

Could you make the small wording replacement? Then it is good to merge from my side.

**Comment by jlprat** [comment]
Oops, will fix de typo later.
And yes, I could try to provide some table with examples in another PR

File: docs/src/main/paradox/java/http/routing-dsl/directives/path-directives/index.md [hunk 2]
```diff
@@ -19,4 +19,16 @@
...
+## Overview of path directives
+
+This is a tiny overview for some of the most common path directives:
+
+* `rawPathPrefix(x)`: it matches x and leaves a suffix (if any) unmatched.
+* `pathPrefix(x)`: is equivalent to `rawPathPrefix(slash().concat(segment(x)))`. It matches a leading slash followed by _x_ and then leaves a suffix unmatched.
+* `path(x)`: is equivalent to `rawPathPrefix(slash().concat(segment(x)).concat(pathEnd()))`. It matches a leading slash followed by _x_ and then the end.
+* `pathEnd`: is equivalent to `rawPathPrefix(pathEnd())`. It is matched only when there is nothing left to match from the path. This directive should not be used at the root as the minimal path is the single slash.
+* `pathSingleSlash`: is equivalent to `rawPathPrefix(slash().concat(pathEnd()))`. It matches when the remaining path is just a single slash.
+* `pathEndOrSingleSlash`: is equivalent to `rawPathPrefix(pathEnd())` or `rawPathPrefix(slash().concat(pathEnd()))`. It matches either when there is no remaining path or is just a single slash.
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/routing-dsl/directives/path-directives/index.md]: very nice doc, thanks!

**Comment by ktoso** [comment]
This is great indeed, thanks a lot!
LGTM

**Pull request merged**